### PR TITLE
Removing PCL binary

### LINKED ISSUE #138: Remove PCL support
As of v1.3.0, the assemblies contained in the `Microsoft.Azure.Devices.Client` nupkg is:

```
- net45
  |- Microsoft.Azure.Devices.Client.dll
- netstandard1.3
  |- Microsoft.Azure.Devices.Client.dll
- uap10.0
  |- Microsoft.Azure.Devices.Client.UWP.dll
- portable-net45+wp8+wpa81+win8+MonoAndroid10+MonoTouch10+Xamarin.iOS10+UAP10
  |- Microsoft.Azure.Devices.Client.PCL.dll
```

**Repro**

Because these assemblies are named differently across different profiles, it breaks runtime resolution in a few cases. The most common would be:

1. Create a .NET Standard 1.3 library that references `Microsoft.Azure.Devices.Client`
2. Reference in a .NET Framework 4.5 application **[Works]**
3. Reference in a UWP application **[Does not work]**

This is because the API name is a combination of the assembly name and the fully qualified API (namespace+class).

**What needs to be done?**

The assembly names should be reconciled to all be `Microsof.Azure.Devices.Client.dll` so that runtime resolution works as expected, especially in circumstances where a .NET Standard library references the devices library and is then referenced by a UWP or Xamarin project.

### PR DESCRIPTION
**Pull request by CIPop**
* Removing PCL binary
* Upgrading Newtonsoft.Json to v10
* Upgrading Android, iOS and UWP samples to work with VS2017, latest Xamarin

Fixes #138

### CODE REVIEW

File: service/Microsoft.Azure.Devices/Microsoft.Azure.Devices.csproj [hunk 1]
```diff
@@ -68,9 +68,8 @@
       <HintPath>$(SolutionDir)\packages\Microsoft.Owin.3.0.1\lib\net45\Microsoft.Owin.dll</HintPath>
       <Private>True</Private>
     </Reference>
-    <Reference Include="Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
-      <HintPath>$(SolutionDir)\packages\Newtonsoft.Json.6.0.8\lib\net45\Newtonsoft.Json.dll</HintPath>
-      <Private>True</Private>
+    <Reference Include="Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
+      <HintPath>..\..\e2e\packages\Newtonsoft.Json.10.0.3\lib\net45\Newtonsoft.Json.dll</HintPath>
```
Comment (jasmineymlo) [inline service/Microsoft.Azure.Devices/Microsoft.Azure.Devices.csproj]: >e2e [](start = 22, length = 3)

wrong path?

File: shared/Microsoft.Azure.Devices.Shared/Microsoft.Azure.Devices.Shared.csproj [hunk 2]
```diff
@@ -41,9 +41,8 @@
     <CodeAnalysisRuleSet>MinimumRecommendedRules.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
-      <HintPath>$(SolutionDir)\packages\Newtonsoft.Json.6.0.8\lib\net45\Newtonsoft.Json.dll</HintPath>
-      <Private>True</Private>
+    <Reference Include="Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
+      <HintPath>..\packages\Newtonsoft.Json.10.0.3\lib\net45\Newtonsoft.Json.dll</HintPath>
```
Comment (jasmineymlo) [inline shared/Microsoft.Azure.Devices.Shared/Microsoft.Azure.Devices.Shared.csproj]: >>..\ [](start = 15, length = 4)

Is this equivalent to $(SolutionDir)?

File: shared/iothub_csharp_shared.sln [hunk 3]
```diff
@@ -107,14 +81,14 @@ Global
...
-		{91DFB837-D8A3-4F54-AE0D-45C95ACD0C2A}.Release_Delay_Sign|Any CPU.Build.0 = Release_Delay_Sign|Any CPU
-		{91DFB837-D8A3-4F54-AE0D-45C95ACD0C2A}.Release_Delay_Sign|ARM.ActiveCfg = Release_Delay_Sign|Any CPU
-		{91DFB837-D8A3-4F54-AE0D-45C95ACD0C2A}.Release_Delay_Sign|ARM.Build.0 = Release_Delay_Sign|Any CPU
-		{91DFB837-D8A3-4F54-AE0D-45C95ACD0C2A}.Release_Delay_Sign|x64.ActiveCfg = Release_Delay_Sign|Any CPU
-		{91DFB837-D8A3-4F54-AE0D-45C95ACD0C2A}.Release_Delay_Sign|x64.Build.0 = Release_Delay_Sign|Any CPU
-		{91DFB837-D8A3-4F54-AE0D-45C95ACD0C2A}.Release_Delay_Sign|x86.ActiveCfg = Release_Delay_Sign|Any CPU
-		{91DFB837-D8A3-4F54-AE0D-45C95ACD0C2A}.Release_Delay_Sign|x86.Build.0 = Release_Delay_Sign|Any CPU
+		{91DFB837-D8A3-4F54-AE0D-45C95ACD0C2A}.Release_Delay_Sign|Any CPU.ActiveCfg = Release|Any CPU
+		{91DFB837-D8A3-4F54-AE0D-45C95ACD0C2A}.Release_Delay_Sign|Any CPU.Build.0 = Release|Any CPU
+		{91DFB837-D8A3-4F54-AE0D-45C95ACD0C2A}.Release_Delay_Sign|ARM.ActiveCfg = Release|Any CPU
```
Comment (jasmineymlo) [inline shared/iothub_csharp_shared.sln]: >Release [](start = 76, length = 7)

not sure if this is intended?  and why?

File: service/Microsoft.Azure.Devices.Uwp/Microsoft.Azure.Devices.Uwp.csproj [hunk 4]
```diff
@@ -11,11 +11,12 @@
     <AssemblyName>Microsoft.Azure.Devices</AssemblyName>
     <DefaultLanguage>en-US</DefaultLanguage>
     <TargetPlatformIdentifier>UAP</TargetPlatformIdentifier>
-    <TargetPlatformVersion>10.0.10240.0</TargetPlatformVersion>
-    <TargetPlatformMinVersion>10.0.10240.0</TargetPlatformMinVersion>
+    <TargetPlatformVersion>10.0.15063.0</TargetPlatformVersion>
+    <TargetPlatformMinVersion>10.0.15063.0</TargetPlatformMinVersion>
```
Comment (jasmineymlo) [inline service/Microsoft.Azure.Devices.Uwp/Microsoft.Azure.Devices.Uwp.csproj]: >15063 [](start = 35, length = 5)

is this required?

File: service/Microsoft.Azure.Devices/test/Microsoft.Azure.Devices.Api.Test/Microsoft.Azure.Devices.Api.Test.csproj [hunk 5]
```diff
@@ -138,9 +138,8 @@
       <HintPath>$(SolutionDir)\packages\Moq.4.2.1510.2205\lib\net40\Moq.dll</HintPath>
       <Private>True</Private>
     </Reference>
-    <Reference Include="Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
-      <HintPath>$(SolutionDir)\packages\Newtonsoft.Json.6.0.8\lib\net45\Newtonsoft.Json.dll</HintPath>
-      <Private>True</Private>
+    <Reference Include="Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Newtonsoft.Json.10.0.3\lib\net45\Newtonsoft.Json.dll</HintPath>
```
Comment (jasmineymlo) [inline service/Microsoft.Azure.Devices/test/Microsoft.Azure.Devices.Api.Test/Microsoft.Azure.Devices.Api.Test.csproj]: >>..\..\..\ [](start = 15, length = 10)

$(SolutionDir)??

**Comment by CIPop** [comment]
CI is failing for this PR in the Android solution build with #197.

File: service/Microsoft.Azure.Devices/Microsoft.Azure.Devices.csproj (hunk 1 above)
Comment (CIPop) [reply service/Microsoft.Azure.Devices/Microsoft.Azure.Devices.csproj]: These are all caused by automated VS2017 edits. I need to revert and ensure we're still using $(SolutionDir) instead.

File: shared/Microsoft.Azure.Devices.Shared/Microsoft.Azure.Devices.Shared.csproj (hunk 2 above)
Comment (CIPop) [reply shared/Microsoft.Azure.Devices.Shared/Microsoft.Azure.Devices.Shared.csproj]: These are all caused by automated VS2017 edits. I need to revert and ensure we're still using $(SolutionDir) instead.

File: shared/iothub_csharp_shared.sln (hunk 3 above)
Comment (CIPop) [reply shared/iothub_csharp_shared.sln]: Most likely introduced by the update to the new UWP.

File: service/Microsoft.Azure.Devices.Uwp/Microsoft.Azure.Devices.Uwp.csproj (hunk 4 above)
Comment (CIPop) [reply service/Microsoft.Azure.Devices.Uwp/Microsoft.Azure.Devices.Uwp.csproj]: I'll look more into this - I was getting some odd errors if I wasn't updating. 
Also, we should avoid support of SDKs that are too old to be useful and instead focus on the latest SDKs that can be used with UWP, including the latest Windows IoT Fall Update.

**Pull request closed without merging**